lib/io/hardware_can.cpp

```diff
@@ -0,0 +1,132 @@
+#include "hardware_can.hpp"
+
+#include <stdio.h>
+#include <unistd.h>
+
+#include <net/if.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+
+namespace hyped::io {
+
+std::optional<std::shared_ptr<HardwareCan>> HardwareCan::create(
+  core::ILogger &logger, const std::string &can_network_interface)
+{
+  std::int16_t socket_id = socket(PF_CAN, SOCK_RAW, CAN_RAW);
+  if (socket_id < 0) {
+    logger.log(core::LogLevel::kFatal, "Unable to open CAN socket");
+    return std::nullopt;
+  }
+  const std::uint8_t interface_index = if_nametoindex(can_network_interface.c_str());
```

Comment:
this returns `unsigned int` aka `std::uint32_t` (or just `auto`)

https://www.man7.org/linux/man-pages/man3/if_nametoindex.3.html

don't just make up types!